### ISSUE TITLE
new rule: don't use `typing.runtime_checkable`

### ISSUE THREAD

**Issue by KotlinIsland** [astral-sh/ruff#21841]
### Summary

this feature isn't sound, and is often not what the user expects. i have seen people fall for it

```py
from typing import Protocol, runtime_checkable


@runtime_checkable
class P(Protocol):
    a: int

class A:
    a: str = "a"


def f(a: object):
    if isinstance(a, P):
        print(a.a + 1)  # WHAT!

f(A())
```

things like pydantic, beartype and optype do a better job at runtime validation than the stdlib, but people would probably assume that the stdlib works properly

**Comment by ntBre** [comment]
I think for a blanket ban like this, [banned-api (TID251)](https://docs.astral.sh/ruff/rules/banned-api) might work well.

https://play.ruff.rs/49f830d7-8d90-4120-8493-20779d25b76d

**Comment by KotlinIsland** [comment]
sure, this is good. but ruff rules are about discoverability, it doesn't help the 99% of people that don't know that `runtime_checkable` doesn't work properly
